biogeochem/EDPhysiologyMod.F90

```diff
@@ -2247,43 +2247,57 @@ subroutine fragmentation_scaler( currentPatch, bc_in)
 
     catanf(t1) = 11.75_r8 +(29.7_r8 / pi) * atan( pi * 0.031_r8  * ( t1 - 15.4_r8 ))
     catanf_30 = catanf(30._r8)
     
     ifp = currentPatch%patchno 
 
-    if ( .not. use_century_tfunc ) then
-    !calculate rate constant scalar for soil temperature,assuming that the base rate constants 
-    !are assigned for non-moisture limiting conditions at 25C.
-      if (bc_in%t_veg24_pa(ifp)  >=  tfrz) then
-        t_scalar = q10_mr**((bc_in%t_veg24_pa(ifp)-(tfrz+25._r8))/10._r8)
-                 !  Q10**((t_soisno(c,j)-(tfrz+25._r8))/10._r8)
-      else
-        t_scalar = (q10_mr**(-25._r8/10._r8))*(q10_froz**((bc_in%t_veg24_pa(ifp)-tfrz)/10._r8))
-                  !Q10**(-25._r8/10._r8))*(froz_q10**((t_soisno(c,j)-tfrz)/10._r8)
-      endif
+    ! Use the hlm temp and moisture decomp fractions by default
+    if ( use_hlm_soil_scalar ) then
+      
+      ! Calculate the fragmentation_scaler
+      currentPatch%fragmentation_scaler =  min(1.0_r8,max(0.0_r8,bc_in%t_scalar_sisl * bc_in%t_scalar_sisl))
+
     else
-      ! original century uses an arctangent function to calculate the 
-      ! temperature dependence of decomposition      
-      t_scalar = max(catanf(bc_in%t_veg24_pa(ifp)-tfrz)/catanf_30,0.01_r8)
-    endif    
-   
-    !Moisture Limitations   
-    !BTRAN APPROACH - is quite simple, but max's out decomp at all unstressed 
-    !soil moisture values, which is not realistic.  
-    !litter decomp is proportional to water limitation on average... 
-    w_scalar = sum(currentPatch%btran_ft(1:numpft))/real(numpft,r8)
+    
+      if ( .not. use_century_tfunc ) then
+      !calculate rate constant scalar for soil temperature,assuming that the base rate constants 
+      !are assigned for non-moisture limiting conditions at 25C.
+         if (bc_in%t_veg24_pa(ifp)  >=  tfrz) then
+         t_scalar = q10_mr**((bc_in%t_veg24_pa(ifp)-(tfrz+25._r8))/10._r8)
+                  !  Q10**((t_soisno(c,j)-(tfrz+25._r8))/10._r8)
+         else
+         t_scalar = (q10_mr**(-25._r8/10._r8))*(q10_froz**((bc_in%t_veg24_pa(ifp)-tfrz)/10._r8))
+                     !Q10**(-25._r8/10._r8))*(froz_q10**((t_soisno(c,j)-tfrz)/10._r8)
+         endif
+      else
+         ! original century uses an arctangent function to calculate the 
+         ! temperature dependence of decomposition      
+         t_scalar = max(catanf(bc_in%t_veg24_pa(ifp)-tfrz)/catanf_30,0.01_r8)
+      endif    
+    
+      !Moisture Limitations   
+      !BTRAN APPROACH - is quite simple, but max's out decomp at all unstressed 
+      !soil moisture values, which is not realistic.  
+      !litter decomp is proportional to water limitation on average... 
+      w_scalar = sum(currentPatch%btran_ft(1:numpft))/real(numpft,r8)
+
+    ! Calculate the fragmentation_scaler
+      allocate(currentPatch%fragmentation_scaler(size(bc_in%t_scalar_sisl)))
+      currentPatch%fragmentation_scaler(:) =  min(1.0_r8,max(0.0_r8,t_scalar * w_scalar))
```

Comment:
why is it now an array?

Reply:
The hlm `t_scalar` and `w_scalar` inputs are indexed by soil level.  Talking to @ckoven we decided to utilize this to enable a `fragmentation_scaler` calculation for each soil level, noting that the below ground litter variables are being indexed by soil level already.  This way the below ground levels will each have a specific fragmentation value.

As you noted in our phone call, we could also investigate the averaging of the temperature and moisture scalars across the different levels.